contracts/StringParserAssemblyLibrary.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+library StringParserAssemblyLibrary {
+
+    function stripAfter(string memory input, string memory symbol) public pure returns (string memory) {
```

Comment:
It's better to implement more logic with assembly than just one or two line expressions throughout the code. There is to few sense for it even for educational purposes.
       -1 point